src/engine/source/json/src/json.cpp

```diff
@@ -1005,8 +1007,7 @@ void Json::merge(rapidjson::Value& source, std::string_view path)
                 }
                 else
                 {
-                    throw std::runtime_error(
-                        "JSON elements must be either objects or arrays to be merged");
+                    throw std::runtime_error("JSON elements must be either objects or arrays to be merged");
```

Comment:
Both elements must be objects or both must be arrays, slightly different, but it could shed some light on the matter.